clang/test/SemaSYCL/array-num-elements-exceeded.cpp

```diff
@@ -0,0 +1,19 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir-unknown -verify -fsyntax-only %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown -verify -fsyntax-only %s
+
+const double big[67000] = {1.0, 2.0, 3.0};
+const int zero_init_array[67000] = {};
```

Comment:
another good repro is:

const int zero_init_array2[67000] = {0,0,0,constexpr_fun_that_returns_0()};

Reply:
> If all you care about is whether the array is default initialized 

We care whether `zeroinitializer` for the array will be generated in LLVM IR. Is there a good method to detect such case on the Sema level?

Reply:
Yeah, you'd have to detect the actual initializers.  See: https://godbolt.org/z/Yo5bcx

You probably ALSO care about this case: https://godbolt.org/z/v8MKrM

Note that the default constructed "S2" is undef, and I'm not particularly sure what differentiates that.  For example, a defaulted constructor on the empty type results in it still being 'undef', but an actual constructor will zero-init and call the constructor at global_var_init: https://godbolt.org/z/fvzK97